### ISSUE TITLE
ExecuTorch MethodMeta incorrectly cached?

### ISSUE THREAD

**Issue by manuelcandales** [pytorch/executorch#14012]
### 🐛 Describe the bug

The following code provides the setup. Basically we defined two models, which run any.dims, one with an empty dim list, and the other with a null dim list. And then we have a "process" function to export and lower to executorch.

```python
import torch
import torch.nn as nn
from executorch.exir import EdgeCompileConfig, to_edge_transform_and_lower
from executorch.extension.pybindings.portable_lib import (
    _load_for_executorch_from_buffer,
)

class AnyDimsModelEmpty(nn.Module):
    def __init__(self):
        super(AnyDimsModelEmpty, self).__init__()

    def forward(self, x):
        return torch.ops.aten.any.dims(x, [], False)

class AnyDimsModelNull(nn.Module):
    def __init__(self):
        super(AnyDimsModelNull, self).__init__()

    def forward(self, x):
        return torch.ops.aten.any.dims(x, None, False)


model_empty = AnyDimsModelEmpty()
model_null = AnyDimsModelNull()


def process(model):
    print('model:', model)
    x = torch.ones([2, 3])

    # Print the output shape
    model_output = model(x)
    print('model output shape:', model_output.shape)

    # Export the model using torch.export
    exported_program = torch.export.export(model, (x,))

    compile_config = EdgeCompileConfig(_check_ir_validity=False)
    et_program = to_edge_transform_and_lower(
        exported_program, compile_config=compile_config
    ).to_executorch()

    executorch_buffer = et_program.buffer

    # Load ExecuTorch model
    executorch_module = _load_for_executorch_from_buffer(executorch_buffer)

    method_meta = executorch_module.method_meta("forward")
    print(f"Method metadata: {method_meta}")
    print()
```

If we only process one model, then everything is fine, but if we process both models, one after the other, the expected output shape is incorrect for the second model. It doesn't matter which one we process first and which one we process second. I think this must be related to some sort of caching?

```python
process(model_empty)
process(model_null)
```
terminal output:
```
model: AnyDimsModelEmpty()
model output shape: torch.Size([2, 3])
Method metadata: MethodMeta(name='forward', num_inputs=1, input_tensor_meta=['TensorInfo(sizes=[2, 3], dtype=Float, is_memory_planned=True, nbytes=24)'], num_outputs=1, output_tensor_meta=['TensorInfo(sizes=[2, 3], dtype=Bool, is_memory_planned=True, nbytes=6)'])

model: AnyDimsModelNull()
model output shape: torch.Size([])
Method metadata: MethodMeta(name='forward', num_inputs=1, input_tensor_meta=['TensorInfo(sizes=[2, 3], dtype=Float, is_memory_planned=True, nbytes=24)'], num_outputs=1, output_tensor_meta=['TensorInfo(sizes=[2, 3], dtype=Bool, is_memory_planned=True, nbytes=6)'])
```

If we only run `process(model_null)`, we get the correct output shape:
```python
# process(model_empty)
process(model_null)
```
terminal output:
```
model: AnyDimsModelNull()
model output shape: torch.Size([])
Method metadata: MethodMeta(name='forward', num_inputs=1, input_tensor_meta=['TensorInfo(sizes=[2, 3], dtype=Float, is_memory_planned=True, nbytes=24)'], num_outputs=1, output_tensor_meta=['TensorInfo(sizes=[], dtype=Bool, is_memory_planned=True, nbytes=1)'])
```

### Versions

nightly

cc @JacobSzwejbka @angelayi

**Comment by manuelcandales** [comment]
The source of this bug seems to be in torch.export. Filed issue: https://github.com/pytorch/pytorch/issues/162279
